frontend/src/app/core/services/pdf-make.service.ts

```diff
@@ -0,0 +1,305 @@
+import { mergeMap } from 'rxjs/operators';
+import { Establishment } from '@core/model/establishment.model';
+import { TrainingRecordCategory } from '@core/model/training.model';
+import { QualificationsByGroup } from '@core/model/qualification.model';
+import { FormatDate } from '@core/utils/date-util';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PdfMakeService {
+  public imageAssets: Record<string, string> = {};
+
+  constructor(private http: HttpClient) {
+    (<any>pdfMake).addVirtualFileSystem(pdfFonts);
+
+    this.loadImageAssets();
+  }
+
+  blobToBase64 = (blob: Blob): Promise<string> => {
+    const reader = new FileReader();
+    reader.readAsDataURL(blob);
+    return new Promise((resolve) => {
+      reader.onloadend = () => {
+        resolve(reader.result as string);
+      };
+    });
+  };
+
+  public async loadImageAssets() {
+    this.http
+      .get('assets/images/logo.png', { responseType: 'blob' })
+      .pipe(mergeMap((blob) => this.blobToBase64(blob)))
+      .subscribe((base64String: string) => {
+        this.imageAssets.sfcLogo = base64String;
+      });
+  }
+
+  public buildTrainingAndQualificationPdfDefinition(
+    workplace: Establishment,
+    mandatoryTraining: TrainingRecordCategory[],
+    nonMandatoryTraining: TrainingRecordCategory[],
+    qualificationsByGroup: QualificationsByGroup,
+    worker,
+    lastUpdatedDate,
+  ): TDocumentDefinitions {
+    // build content using helper sections
+    const contentBody = [
+      this.sectionHeader('Training and qualifications'),
+      this.staffInfoSection(workplace, worker, lastUpdatedDate),
+      ...this.mandatoryTrainingSection(mandatoryTraining),
+      ...this.nonMandatoryTrainingSection(nonMandatoryTraining),
+      ...this.qualificationSection(qualificationsByGroup),
+    ];
+
+    return {
+      pageSize: 'A4',
+      pageOrientation: 'portrait',
+      pageMargins: [40, 60, 40, 40],
+
+      header: (currentPage = 1, pageCount = 1): Content => ({
+        columns: [
+          {
+            text: `Page ${currentPage} of ${pageCount}`,
+            alignment: 'right',
+            margin: [20, 30] as [number, number],
+          },
+        ],
+      }),
+
+      content: [
+        {
+          columns: [
+            { image: 'sfcLogo', alignment: 'left' },
+            {
+              table: {
+                body: [
+                  [
+                    {
+                      text: 'Adult Social Care Workforce Data Set',
+                      bold: true,
+                      color: '#ffffff',
+                    },
+                  ],
+                ],
+              },
+              layout: {
+                fillColor: '#1d70b8',
+                paddingTop: () => 5,
+                paddingBottom: () => 5,
+                hLineWidth: () => 0,
+                vLineWidth: () => 0,
+              },
+              width: '70%',
+              margin: [5, 20, 0, 20] as [number, number, number, number],
```

Comment:
```suggestion
              margin: [5, 20, 0, 20] as Margins,
```
We can replace these [number, number, number, number] with the `Margins` type from the pdfmake library 

```
import { Content, Margins, TDocumentDefinitions } from 'pdfmake/interfaces';
```